envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/clusters/EnvoyClustersFactory.kt

```diff
@@ -393,7 +477,7 @@ class EnvoyClustersFactory(
                                         )
                                 )
                     }
-                ).setServiceName(clusterConfiguration.serviceName)
```

Comment:
[link to documentation](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/endpoint/v3/endpoint.proto.html#config-endpoint-v3-clusterloadassignment)
Name of the cluster. This will be the [service_name](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/cluster/v3/cluster.proto#envoy-v3-api-field-config-cluster-v3-cluster-edsclusterconfig-service-name) value if specified in the cluster [EdsClusterConfig](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/cluster/v3/cluster.proto#envoy-v3-api-msg-config-cluster-v3-cluster-edsclusterconfig).

So it overrides clusterName specified in a cluster and it's optional, and cluster name is specified in all cases, so I think it should be safe to remove it, otherwise I have to put here this additional logic with cluster name